source/common/network/connection_impl.cc

```diff
@@ -235,13 +242,22 @@ void ConnectionImpl::onFileEvent(uint32_t events) {
     return;
   }
 
+  if (events & Event::FileReadyType::Closed) {
+    // We never ask for both early close and read at the same time. If we are reading, we want to
+    // consume all available data.
+    ASSERT(!(events & Event::FileReadyType::Read));
+    conn_log_debug("remote early close", *this);
+    closeSocket(ConnectionEvent::RemoteClose);
+    return;
+  }
+
   // Read may become ready if there is an error connecting. If still connecting, skip straight
   // to write ready which is where the connection logic is.
   if (!(state_ & InternalState::Connecting) && (events & Event::FileReadyType::Read)) {
     onReadReady();
   }
 
-  // Possible for a read event close the socket.
+  // Possible for a read event to close the socket.
```

Comment:
Is this comment misplaced? I'm not sure what it has to do with the onWriteReady() case.